src/cfnlint/transform.py

```diff
@@ -70,6 +71,13 @@ def _replace_local_codeuri(self):
             if resource_type == 'AWS::Serverless::Function':
 
                 Transform._update_to_s3_uri('CodeUri', resource_dict)
+                auto_publish_alias = resource_dict.get('AutoPublishAlias')
+                if isinstance(auto_publish_alias, dict):
+                    if len(auto_publish_alias) == 1:
+                        for k, v in auto_publish_alias.items():
+                            if k == 'Ref':
```

Comment:
Do we need to handle !Join and !Sub as well?  Might be better to think of this as a general function to populate fake values for all params in a resource (yeah.... I know...)

Reply:
From the SAM Docs.
`
AutoPublishAlias             Ref of a CloudFormation Parameter  Alias resources created by SAM uses a LocicalId <FunctionLogicalId+AliasName>. So SAM either needs a string for alias name, or a Ref to template Parameter that SAM can resolve into a string.
`

Reply:
@cmmeyer is right. This fix shouldn't be specific to `AutoPublishAlias`. My expectation was the fix should put dummy values in for any required template parameters (no default specified). That will future-proof this against other similar issues.